test/e2e_node/pods_lifecycle_termination_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+const (
+	PodSigkillTimout = 5 * time.Minute
+)
+
+// Pod sigkill test will cover pods with graceful termination period set but failed
+// to terminate and forcefully killed by kubelet. This test exaimine pod's container's
+// exit code is 137 and the exit reason is `Error`
+var _ = SIGDescribe("Pod sigkill [LinuxOnly] [NodeConformance]", func() {
+	f := framework.NewDefaultFramework("sigkill-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	containerName := "sigkill-target-container"
+	podSpec := getSigkillTargetPod("sigkill-target-pod", containerName)
+	runSigkillTest(f, podSpec)
+})
+
+func runSigkillTest(f *framework.Framework, podSpec *v1.Pod) {
+	ginkgo.Context("", func() {
+		ginkgo.BeforeEach(func() {
+			ginkgo.By("setting up the pod to be used in the test")
+			e2epod.NewPodClient(f).Create(context.TODO(), podSpec)
+		})
+
+		ginkgo.It("The containers terminated forcefully by Sigkill should have the correct exit code(137) and reason (Error)", func() {
+
+			ginkgo.By("Waiting for the pod to be running")
+			err := e2epod.WaitForPodNameRunningInNamespace(context.TODO(), f.ClientSet, podSpec.Name, f.Namespace.Name)
+			framework.ExpectNoError(err, "Failed to await for the pod to be running: %q", podSpec.Name)
+
+			ginkgo.By(fmt.Sprintf("Deleting the pod (%v/%v) to set a deletion timestamp", f.Namespace.Name, podSpec.Name))
+			// wait a little bit to make sure the we are inside the while and that the trap is registered
+			time.Sleep(time.Second)
```

Comment:
what "while"?

Reply:
+1, I think this can be dropped for this test.

Reply:
I tried removing this sleep, and saw flaky failure once during test. It's better to keep it here.
Revised the comment.